### ISSUE TITLE
Blockchain sync failure

### ISSUE THREAD

**Issue by bol-van** [bitcoin/bitcoin#6606]
I'm on Bitcoin core v0.11.0 windows x64.
OS is Windows Server 2012 R2.

I've been using bitcoin core for years without significant problems, but last month something happened. Database got corrupted. I tried to delete all but wallet.dat, resync database. Tried ~5 times, put datadir to different hard drives. At random position sync stops with error. After process relaunch same error is displayed and program crashes with assertion.

![bitcoin_read_database](https://cloud.githubusercontent.com/assets/9076680/9578556/b2ccd616-4ff3-11e5-8850-832cabdd563f.jpg)
![1](https://cloud.githubusercontent.com/assets/9076680/9578580/dfaa8002-4ff3-11e5-8f54-3e8d53028fbe.jpg)
![2](https://cloud.githubusercontent.com/assets/9076680/9578579/dfaa22ba-4ff3-11e5-9054-62f75516ca00.jpg)

**Comment by bol-van** [comment]
Same thing happens to bitcoind.

C:\Program Files\Bitcoin\daemon>bitcoind.exe  -datadir=H:\home\k\bitcoin
Error: Error reading from database, shutting down.

This application has requested the Runtime to terminate it in an unusual way.
Please contact the application's support team for more information.

debug.log :

2015-08-31 12:39:34 LevelDB read failure: Corruption: block checksum mismatch
2015-08-31 12:39:34 Corruption: block checksum mismatch
2015-08-31 12:39:34 Error: Error reading from database, shutting down.
2015-08-31 12:39:34 Error reading from database: Database corrupted

**Comment by bol-van** [comment]
Is it all fork related ?

**Comment by laanwj** [comment]
"Error reading from database: Database corrupted" levelDB corruption is usually caused by disk or memory corruption (while writing to disk).
You could try using `-par=1` to restrict syncing to one thread. Sometimes this helps when, for example, the CPU is overheating.

**Comment by bol-van** [comment]
Unlikely this is RAM or DISK problem. OS runs for weeks without problems, memtest report nothing.
No bad block events in the event log. One of the disk I tried to put db on is several days old.
Will try par=1 as you suggested.
Any ways to further diagnose the source of the problem ?

**Comment by bol-van** [comment]
I reproduced exact same behavior in VM with Windows Server 2003 X64.
Pls someone try to resync the whole db ! Am I alone with this ?

**Comment by laanwj** [comment]
I'd be interested to know if the same happens in that VM with Bitcoin 10.2.

**Comment by wtogami** [comment]
You are able to reproduce the failure on other hardware?
What about bitcoind or bitcoin-qt for Linux in a VM?

**Comment by bol-van** [comment]
Additional notice.

Both 0.10.2 and 0.11.0 cannot start db sync when datadir is on "\\vmhost\shared folders" and successfully do when datadir is on windows network drive

**Comment by bol-van** [comment]
I have one guess. Trouble can be in memory mapped files.
I also run BURST coin pocminer. It extensively uses mapped files. Because of that kernel paged pool grows very large - up to more than half of the physical memory (its gigabytes). Huge pooltag is "MmSt". Detailed subject description is here : http://blogs.technet.com/b/askperf/archive/2011/09/23/getting-to-know-the-mmst-pool-tag.aspx
I'm on 24 GB system and set the PoolUsageMaximum to 10 (its 10 percent of RAM, 2.4G in my case). This measure effectively limit MmSt growth and it worked great until... what changed in last weeks ?
I replaced hard drive which contain BURST miner plots. This time I formatted NTFS volume with 64K cluster size (was 4K).
And probably from that point bitcoin db corruptions started.
Now i killed pocminer and trying to sync bitcoin both on the host and in VM.
Without pocminer bitcoin could start sync on vmware-host shared folder.
Will report after my guess is confirmed or no.

**Comment by bol-van** [comment]
Yes, trouble is in BURST miner. Without it sync was successful.

**Comment by laanwj** [comment]
Thanks for looking into this so deeply. This issue could be useful for other people that experience issues on windows.

I still wonder how the combination of hw and sw caused corruption, but it's likely the problem lies outside bitcoin core if it affects other software negatively as well.

**Comment by bol-van** [comment]
One of the negative effects was the following.
Attempts to start db sync from bitcoin core running in vmware guest on vmware host drive were failing just at the start. Then I tried to mount network drive from vm guest to vm host using virtual network (regular 'net use \\192.168.1.5') and sync db to that drive.
After some time I saw messages in the tray stating that windows could not flush data to network drive and data could be lost. Obviously, bitcoin core cannot display such messages, explorer.exe displays them. Event source is lanman redirector running in the kernel of the guest OS not being able to successfully get confirmation from the server side.  Thus the lanmanserver (The 'Server' service) component on the host was experiencing problems in exhausting paged pool condition.
Its all very strange because in the task manager on the host I see that pool is being trimmed from 2.8G to 800M after its exhaustion and then again grows to 2.8G.
I can suppose some paged pool allocations fail before MmSt trim actually happens.
Windows architecture problem ? I know burst miner is badly designed. It should not map terabytes of data files to memory. But also OS should not behave bad in this condition. If MmSt pool is like cache it must be trimmed transparently without alloc fails.
